google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -625,7 +627,12 @@ Builder setCustomerEncryption(CustomerEncryption customerEncryption) {
 
     @Override
     Builder setKmsKeyName(String kmsKeyName) {
-      this.kmsKeyName = kmsKeyName;
+      String cryptoKeyVersions = "/cryptoKeyVersions/";
+      if (kmsKeyName != null && kmsKeyName.contains(cryptoKeyVersions)) {
+        this.kmsKeyName = kmsKeyName.substring(0, kmsKeyName.indexOf(cryptoKeyVersions));
```

Comment:
Update to, this is what we did in Python to prevent unintended behavior. 

```java
this.kmsKeyName = "";
```

Reply:
done